src/execution/lazy/store.rs

```diff
@@ -78,10 +78,8 @@ impl LazyStore {
         exec: &mut EvaluationContext,
     ) -> Result<graph::Value, ExecutionError> {
         let variable = &self.elements[variable.store_location];
-        let debug_info = variable.debug_info;
-        let value = variable.force(exec).with_context(|| {
-            format!("via {} at {}", (*variable.state.borrow()), debug_info).into()
-        })?;
+        let debug_info = variable.debug_info.clone();
+        let value = variable.force(exec).with_context(|| debug_info.0.into())?;
```

Comment:
Here when a value is loaded from the store. Since values originate from other statements (the ones where they were bound), this ensures that an evaluation error reports on the statement that constructed the value, not the statement that forced its evaluation.